### ISSUE TITLE
Clarification on TSN Flow Extraction Time & Storage Requirements for FG2021-BoLD

### ISSUE THREAD

**Issue by ghost** [GiannisPikoulis/FG2021-BoLD#5]
Hi FG2021-BoLD team,

I’m currently working with your FG2021-BoLD implementation and wanted to confirm whether the preprocessing requirements I’m seeing are expected.

To generate the TSN RGB + flow inputs, I’ve been using a slightly modified extraction pipeline that runs on Windows (for convenience), based on your extract_flow.py and the standard TSN frame extraction logic. The script iterates over the BoLD videos/003 directory and outputs:

TSN RGB frames at 340×256 resolution

TSN Optical Flow frames (TV-L1), stored as JPEG pairs per frame

Even with 8 workers, the extraction time appears extremely long — on a modern RTX GPU machine, the TV-L1 flow computation is still CPU-bound and seems like it could take several days to weeks for the full dataset.

Additionally, the storage requirement for stride-1 extraction seems very large. Based on partial extraction, it looks like the full RGB + TV-L1 flow set might exceed 600 GB.

Before I commit to a full preprocessing run, I wanted to confirm:

Are these extraction times expected when generating TSN RGB + TV-L1 flow for the entire BoLD dataset?

Is the 500–700 GB storage footprint typical, or are there recommended compression/format alternatives that you used in your experiments?

Is there an officially endorsed preprocessing pipeline (Windows or Linux) that you recommend for speed or efficiency?

Here is the command structure I’ve been using for extraction (Windows PowerShell):

python extract_tsn_rgb_flow_bold.py \
  --videos_root "<path to BoLD videos>" \
  --rgb_root "<output tsn_rgb>" \
  --flow_root "<output tsn_flow>" \
  --resize_w 340 --resize_h 256 \
  --flow_algo tvl1 \
  --stride 1 \
  --num_workers 8

Script is attached.

I want to ensure that I’m aligning with your intended preprocessing setup, especially before running training or inference using your pretrained checkpoints.

Any guidance or confirmation you can share would be greatly appreciated. Thank you for making this work available, it’s extremely valuable for emotion recognition research.

[extract_tsn_rgb_flow_bold.py](https://github.com/user-attachments/files/23980608/extract_tsn_rgb_flow_bold.py)

**Comment by GiannisPikoulis** [comment]
Hello.

First of all thanks for reaching out. It has been a long time since I touched this repo (~4 years). As far as I can remember, I never really questioned or modified the original flow computation proces presented in https://github.com/yjxiong/temporal-segment-networks/tree/master, because I never had the time to; I just took it as a given and built on top of it. The storage requirements, I believe will be close to 1 TB in total for the whole project. I believe the best thing you can do to check whether you are going in the right direction is to compute RGB+Flow for a small segment of the test set, then grab my checkpoints and run inference on it. If you get reasonable outputs, then move on with the rest of your research as usual.

Kind regards,
Ioannis Pikoulis.

**Comment by ghost** [comment]
Thanks so much for your reply. Before I upgrade my hard drive could you
please share with me (at least a ball park) of class performance results
for bold+rgb and/or fusion?
